### DISCUSSION TITLE
basic_dev_environment

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/labview_book#36] [Announcements]
# 按自己的喜好设置编程环境

https://lv.qizhen.xyz/basic_dev_environment

**Comment by NavySSD** [comment]
👍

**Comment by Lyz-623** [comment]
太赞了

**Comment by ruanqizhen** [comment]
之前博客上的相关文章，搬到这里：

### Palette API
对函数和控件面板的调整，并不是只有手工调整着一个方法。LabVIEW 8.6 中提供了在程序中修改函数和控件面板的 API。通过这些 API VI，用户就可以在程序里读取或设置 .mnu 文件里的内容，从而通过编程来改变函数和控件面板。

**Comment by ruanqizhen** [comment]
之前博客上的相关文章，搬到这里：

### 鼠标配合Ctrl键的妙用
 2009年05月22日 1 Minutes
LabVIEW中有众多的快捷键，这些快捷键可以大大加快编程速度。不过记住这些快捷键本身也是个挑战，比如说我吧，我只能记住几个最常用的快捷键，比如 Ctrl+C，Ctrl+V 等。我编程序主要还是依赖鼠标，单一个鼠标应付各种工作还是显得有些吃力的，编程速度可能没有熟练使用快捷键那么迅速。编的程序多了，不经意间就发现了一些鼠标配合Ctrl键的妙用，于是就把最有用的几种用法记录下来了。

复制对象
在前面板或程序框图上选中一个或多个对象（控件、接线端、函数等），然后拖动鼠标到另一个位置。这时如果按下Ctrl键，光标的箭头旁就会出现一个加号，表示这时的操作变成的复制，而不再是移动对象了。先松开鼠标，再松开Ctrl键，就会在光标的新位置上放置一个选中对象的副本。这种复制对象的方法比使用 Ctrl+C，Ctrl+V 还便捷。这个方法还可以复制一些使用 Ctrl+C，Ctrl+V 无法复制的对象，比如对控件的属性节点和调用节点。


图1：按下Ctrl键，用鼠标拖动一个对象


图2：生成了被拖动对象的一个副本

先按下Ctrl键，再按鼠标选中一个对象，然后拖动它，也可复制这个对象。先按下Shift键，再按鼠标选中一个对象，然后拖动它，也有特定的功能。它可以使得对象只在完全水平或者垂直的方向上移动一个对象。如果同时按下Ctrl和Shift键，再按鼠标选中一个对象，然后拖动它，就是把前两项功能合用：在当前对象的正上下或左右方复制一份新对象，这有利于直接编写出整洁的代码。

拓展程序框图空间
有时候编着编着程序，就发现，需要在当前程序框图的某处插入一些额外的代码，而这里已经挤满的节点，在没有额外空间容纳新节点了。这时，可以按下Ctrl键，再在需要放置新代码的地方按下鼠标，拖动鼠标拉出一块矩形区域。松开鼠标，程序框图会自动把代码向周围拖动，空出这块区域用于放置新代码。

比如图3中的程序，需要在程序最中央位置，“Play One”常量的左侧腾出一块控件放置新代码。这时，按下Ctrl键，再在这一位置点击鼠标左键，然后拖动，拉出合适大小的一个方块。松开鼠标，这个方块的位置就空出来了，原来的代码被挤到旁边去了（图4）。


图3：按下Ctrl键，用鼠标在需要插入代码的地方拖出一个方块


图4：腾出空间后的程序框图

直接打开子VI的程序框图
在程序框图上，鼠标双击一个子VI，会打开这个VI的前面板。实际上，对于子VI来说，我们通常不关心它的前面板，而是需要查看它的程序框图。如果按住Ctrl键，再双击程序框图上的一个子VI，就可以直接打开这个子VI的程序框图了。

调换函数输入参数的连线
图5中程序有一个加法函数，它的两个输入数据线相连。但是这两条线出现了交叉，如果能交换一下它们的位置，程序会显得整洁不少。这时，先把鼠标移动到加法函数的输入参数接线端上，再按下Ctrl键，光标会变成一个倒着的“&”符号的形状。松开鼠标，这两个输入端的连线就会交换位置，交换位置后的程序如图6所示。这个方法只对有两个输入参数的函数有效。


图5：准备交换加法函数两个输入数据连线的位置


图6：交换位置后的程序框图

翻看选择结构、事件结构、层叠式顺序结构的页面
以选择结构为例：它每次只能显示一个页面，也就是一个条件处理分支。如果需要查看下一分支或上一分支的代码，可以把光标移动到条件结构内部的空白处，按下Ctrl键，然后滚动鼠标滚轮。每滚一下，条件结构就会切换显示一页。
